contrib/opencensus-ext-azure/opencensus/ext/azure/protocol/__init__.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class Object(dict):
+    def __init__(self, *args, **kwargs):
+        super(Object, self).__init__(*args, **kwargs)
+        for key in kwargs:
+            self[key] = kwargs[key]
+
+    def __repr__(self):
+        tmp = {}
+        while True:
+            for item in self.items():
+                if item[0] not in tmp:
+                    tmp[item[0]] = item[1]
+            if self.prototype == self:
+                break
+            self = self.prototype
+        return repr(tmp)
+
+    def __setattr__(self, name, value):
+        self[name] = value
+
+    def __getattr__(self, name):
+        try:
+            return self[name]
+        except KeyError as ex:
+            raise AttributeError("'{}' object has no attribute {}".format(
+                type(self).__name__,
+                ex,
+            ))
+
+    def __getitem__(self, key):
+        if self.prototype is self:
+            return super(Object, self).__getitem__(key)
+        if key in self:
+            return super(Object, self).__getitem__(key)
+        return self.prototype[key]
+
+
+Object.prototype = Object()
+
+
+class Data(Object):
+    prototype = Object(
+        baseData=None,
+        baseType=None,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(Data, self).__init__(*args, **kwargs)
+        self.baseData = self.baseData
+        self.baseType = self.baseType
+
+
+class Envelope(Object):
+    prototype = Object(
+        ver=1,
```

Comment:
ver=1. Is this typo? Looks like all other datatypes have ver=2.

Reply:
It is indeed 1.

https://github.com/Microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/Envelope.bond#L10